unit-tests/src/test/scala/java/util/FormatterSuite.scala

```diff
@@ -0,0 +1,4001 @@
+      assertEquals(f.locale(), Locale.KOREA)
+      f.close()
+    }
+
+    assertThrows[UnsupportedEncodingException](new Formatter(
+      notExist.getPath(),
+      "ISO 1111-1",
+      Locale.CHINA)) // fails #816
+
+    locally {
+      val f = new Formatter(fileWithContent.getPath(),
+                            "UTF-16BE",
+                            Locale.CANADA_FRENCH)
+      assertEquals(0, fileWithContent.length())
+      f.close()
+    }
+
+    if (!root) {
+      assertThrows[FileNotFoundException](
+        new Formatter(readOnly.getPath(),
+                      Charset.defaultCharset().name(),
+                      Locale.ITALY))
+    }
+  }
+
+  test("Constructor(File)") {
+    // segfault
+    // assertThrows[NullPointerException](new Formatter(null.asInstanceOf[File]))
+
+    locally {
+      val f = new Formatter(notExist)
+      assertEquals(f.locale(), Locale.getDefault())
+      f.close()
+    }
+
+    locally {
+      val f = new Formatter(fileWithContent)
+      assertEquals(0, fileWithContent.length())
+      f.close()
+    }
+
+    if (!root) {
+      assertThrows[FileNotFoundException](new Formatter(readOnly))
+    }
+  }
+
+  testFails("Constructor(File, String)", 816) {
+    // OutputStreamWriter should throw UnsupportedEncodingException (NOT UnsupportedCharsetException)
+
+    // segfault
+    // assertThrows[NullPointerException](new Formatter(null.asInstanceOf[File], Charset.defaultCharset().name()))
+
+    locally {
+      val f = new Formatter(notExist, Charset.defaultCharset().name())
+      assertEquals(f.locale(), Locale.getDefault)
+      f.close()
+    }
+
+    locally {
+      val f = new Formatter(fileWithContent, "UTF-16BE")
+      assertEquals(0, fileWithContent.length)
+      f.close()
+    }
+
+    if (!root) {
+      assertThrows[FileNotFoundException](
+        new Formatter(readOnly, Charset.defaultCharset().name()))
+    }
+
+    // segfault
+    // try {
+    //   assertThrows[NullPointerException](new Formatter(notExist, null))
+    // } finally if (notExist.exists()) {
+    //   // Fail on RI on Windows, because output stream is created and
+    //   // not closed when exception thrown
+    //   assertTrue(notExist.delete())
+    // }
+
+    try {
+      assertThrows[UnsupportedEncodingException](
+        new Formatter(notExist, "ISO 1111-1")) /// fails #816
+    } finally if (notExist.exists()) {
+      // Fail on RI on Windows, because output stream is created and
+      // not closed when exception thrown
+      assertTrue(notExist.delete())
+    }
+  }
+
+  testFails("Constructor(File, String, Locale)", 816) {
+    // OutputStreamWriter should throw UnsupportedEncodingException (NOT UnsupportedCharsetException)
+
+    // segfault
+    // assertThrows[NullPointerException](new Formatter(null.asInstanceOf[File], Charset.defaultCharset().name(), Locale.KOREA))
+
+    // segfault
+    // assertThrows[NullPointerException](new Formatter(notExist, null, Locale.KOREA))
```

Comment:
Dereferencing null is undefined behavior in Scala Native. So segfault is expected.